pkg/admission/plugins.go

```diff
@@ -0,0 +1,103 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/admission/plugin/namespace/lifecycle"
+	"k8s.io/apiserver/pkg/admission/plugin/resourcequota"
+	mutatingwebhook "k8s.io/apiserver/pkg/admission/plugin/webhook/mutating"
+	validatingwebhook "k8s.io/apiserver/pkg/admission/plugin/webhook/validating"
+	kubeapiserveroptions "k8s.io/kubernetes/pkg/kubeapiserver/options"
+	certapproval "k8s.io/kubernetes/plugin/pkg/admission/certificates/approval"
+	certsigning "k8s.io/kubernetes/plugin/pkg/admission/certificates/signing"
+	certsubjectrestriction "k8s.io/kubernetes/plugin/pkg/admission/certificates/subjectrestriction"
+	"k8s.io/kubernetes/plugin/pkg/admission/defaulttolerationseconds"
+	"k8s.io/kubernetes/plugin/pkg/admission/limitranger"
+	"k8s.io/kubernetes/plugin/pkg/admission/network/defaultingressclass"
+	"k8s.io/kubernetes/plugin/pkg/admission/nodetaint"
+	podpriority "k8s.io/kubernetes/plugin/pkg/admission/priority"
+	"k8s.io/kubernetes/plugin/pkg/admission/runtimeclass"
+	"k8s.io/kubernetes/plugin/pkg/admission/security/podsecurity"
+	"k8s.io/kubernetes/plugin/pkg/admission/serviceaccount"
+	"k8s.io/kubernetes/plugin/pkg/admission/storage/persistentvolume/resize"
+	"k8s.io/kubernetes/plugin/pkg/admission/storage/storageclass/setdefault"
+	"k8s.io/kubernetes/plugin/pkg/admission/storage/storageobjectinuseprotection"
+)
+
+// AllOrderedPlugins is the list of all the plugins in order.
+var AllOrderedPlugins = beforeWebhooks(kubeapiserveroptions.AllOrderedPlugins)
+
+func beforeWebhooks(recommended []string, plugins ...string) []string {
+	ret := make([]string, 0, len(recommended)+len(plugins))
+	for _, plugin := range recommended {
+		if plugin == mutatingwebhook.PluginName {
+			ret = append(ret, plugins...)
+		}
+		ret = append(ret, plugin)
+	}
+	return ret
+}
+
+// RegisterAllKcpAdmissionPlugins registers all admission plugins.
+// The order of registration is irrelevant, see AllOrderedPlugins for execution order.
+func RegisterAllKcpAdmissionPlugins(plugins *admission.Plugins) {
+	kubeapiserveroptions.RegisterAllAdmissionPlugins(plugins)
+}
+
+var defaultOnPluginsInKcp = sets.NewString(
+	lifecycle.PluginName,                // NamespaceLifecycle
+	limitranger.PluginName,              // LimitRanger
+	serviceaccount.PluginName,           // ServiceAccount
+	defaulttolerationseconds.PluginName, // DefaultTolerationSeconds
```

Comment:
This only runs against pods. Don't enable?

Reply:
removed.